test/test_tensorexpr.py

```diff
@@ -1262,14 +1264,52 @@ def bias_gelu(bias, y):
             x = bias + y
             return x * 0.5 * (1.0 + torch.erf(x / 1.41421))
 
-
         for device in devices:
             a = torch.rand(1024, dtype=torch.half, device=device)
             b = torch.rand(1024, dtype=torch.half, device=device)
             traced = torch.jit.trace(bias_gelu, (a, b))
             x = warmup_and_run_forward(traced, a, b)
             self.assertLastGraphAllFused()
 
+    def test_half_module_backward(self):
+        devices = ["cuda"] if torch.cuda.is_available() else []
+        dtype = torch.float16
+
+        for device in devices:
+            class HalfModule(nn.Module):
+                def __init__(self, inplanes, planes):
+                    super().__init__()
+
+                    self.fixup_bias2a = nn.Parameter(torch.zeros(1))
+                    self.fixup_scale = nn.Parameter(torch.ones(1))
+                    self.fixup_bias2b = nn.Parameter(torch.zeros(1))
+
+                def forward(self, x):
+                    identity = x
+                    out = x
+
+                    out = out + self.fixup_bias2a
+                    out = out * self.fixup_scale + self.fixup_bias2b
+
+                    return out * out + identity
+
+            net_ref = HalfModule(64, 64).to(dtype=dtype, device=device)
+            net = torch.jit.script(HalfModule(64, 64)).to(dtype=dtype, device=device)
+            inp = torch.randn(16, 64, 16, 16, dtype=dtype, device=device)
+
+            for i in range(10):
+                for param in net.parameters():
+                    param.grad = None
+                for param in net_ref.parameters():
+                    param.grad = None
+
+                net(inp).mean().backward()
```

Comment:
imo it would be nice to rewrite this without calling backward, since autograd is complex & makes this test a little harder to reason abt

Reply:
I think original bug may have required it. The issue didn't appear consistently in the NNC output (which is deterministic) so it's something above this layer.